domain/modelmigration/state/state_test.go

```diff
@@ -56,6 +58,54 @@ func (s *migrationSuite) SetUpTest(c *gc.C) {
 // uuid from the database.
 func (s *migrationSuite) TestGetControllerUUID(c *gc.C) {
 	controllerId, err := New(s.TxnRunnerFactory()).GetControllerUUID(context.Background())
-	c.Check(err, jc.ErrorIsNil)
+	c.Assert(err, jc.ErrorIsNil)
 	c.Check(controllerId, gc.Equals, s.controllerUUID.String())
 }
+
+// TestGetAllInstanceIDs is asserting the happy path of getting all instance
+// IDs for the model.
+func (s *migrationSuite) TestGetAllInstanceIDs(c *gc.C) {
+	// Add two different instances.
+	db := s.DB()
+	machineState := machinestate.NewState(s.TxnRunnerFactory(), loggertesting.WrapCheckLog(c))
+
+	err := machineState.CreateMachine(context.Background(), "666", "0", "deadbeef")
+	c.Assert(err, jc.ErrorIsNil)
+	// Add a reference AZ.
+	_, err = db.ExecContext(context.Background(), "INSERT INTO availability_zone VALUES('deadbeef', 'az-1')")
```

Comment:
**question:** Why it is important for the test to add an availability zone ?

Reply:
It's just because `machine_cloud_instance` table has a foreign key to it, if I don't add one I'll get a failure when creating a row.

Reply:
Thanks for the answer, for me it would have been handy to write:

```go
// Add a reference AZ to fulfill FK constraint on machine_cloud_instance.<whatever column>
```

Add a reference AZ is just not enough context IMHO. By gones ;)